systems/plants/RigidBody.cpp

```diff
@@ -37,23 +37,13 @@ RigidBody::RigidBody(void) :
 	velocity_num_start = 0;
 	body_index = 0;
 	mass = 0.0;
-	floating = 0;
-	pitch = 0;
-	com << Vector3d::Zero(), 1;
+  com = Vector3d::Zero();
 	I << Matrix<double, TWIST_SIZE, TWIST_SIZE>::Zero();
-	T = Matrix4d::Identity();
-	Tdot = Matrix4d::Zero();
-	Ttree = Matrix4d::Identity();
-	T_body_to_joint = Matrix4d::Identity();
 }
 
 void RigidBody::setN(int nq, int nv) {
-  dTdq = MatrixXd::Zero(3*nq,4);
-  dTdqdot = MatrixXd::Zero(3*nq,4);
-  ddTdqdq = MatrixXd::Zero(3*nq*nq,4);
-
   dJdq.resize(J.size(), nq);
-  dTdq_new.resize(T.size(), nq);
+  dTdq_new.resize(HOMOGENEOUS_TRANSFORM_SIZE, nq);
```

Comment:
I've started renaming all of this while extracting out all of the kinematics cache into a KinematicsCache object. Next pull request after the templated DrakeJoint stuff.